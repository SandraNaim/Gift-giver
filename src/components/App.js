import React, { Component } from "react";
import { Button } from "react-bootstrap";

class App extends Component {
  constructor(){
    super();
    this.state = {
      gifts: []
    }
  }

  addGifts = () => {
    const { gifts } = this.state
    const ids = gifts.map((gift) => gift.id)
    const max_id = ids.length > 0 ? Math.max(...ids) : 0
    gifts.push({id: max_id+1})
    this.setState({ gifts })
  }

  render(){
    return(
      <div>
        <h2>Gift Givers</h2>
        <div data-test="gift-list">
          {
            this.state.gifts.map((gift) => {
              return<div key={gift.id}></div>
            })
          }
        </div>
        <Button data-test="add-btn" onClick={this.addGifts}>Add gifts</Button>
      </div>
    )
  }
}

export default App;
