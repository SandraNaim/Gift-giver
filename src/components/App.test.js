import React from "react";
import { shallow } from 'enzyme';
import App from './App';

const app = shallow(<App />);


it('renders correctly', () => {
  expect(app).toMatchSnapshot();
})

it('initialize a state of an empty gifts array', () => {
  expect(app.state().gifts).toEqual([]);
})

it('add new gift after clicking on the button', () => {
  app.find("[data-test='add-btn']").simulate('click');
  expect(app.state().gifts).toEqual([{id: 1}])
})