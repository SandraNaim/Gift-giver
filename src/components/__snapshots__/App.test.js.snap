// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Gift Givers
  </h2>
  <Button
    active={false}
    data-test="add-btn"
    disabled={false}
    onClick={[Function]}
    variant="primary"
  >
    Add gifts
  </Button>
</div>
`;
